crates/docs/src/lib.rs

```diff
@@ -539,6 +554,133 @@ fn render_search_type_ahead<'a, I: Iterator<Item = &'a ModuleDocumentation>>(mod
+                        example_description_question_buf.push_str("```md\n");
+                        example_type_question_buf
+                            .push_str(format!("{}\n", annotation_buf).as_str());
+                        example_description_question_buf.push_str("```\n\n");
+                    }
+
+                    if example_description_question_buf.is_empty() && !description_buf.is_empty() {
+                        example_description_question_buf
+                            .push_str("**Description Question Example**\n\n");
+                        example_description_question_buf.push_str("**Question:**\n");
+                        example_description_question_buf
+                            .push_str(format!("What does `{}` do?\n\n", doc_def.name).as_str());
+                        example_description_question_buf.push_str("**Response:**\n");
+                        example_description_question_buf
+                            .push_str(format!("{}\n\n", description_buf).as_str());
+                        example_description_question_buf.push_str("**Source:**\n");
+                        example_description_question_buf.push_str("```md\n");
+                        example_description_question_buf
+                            .push_str(format!("{}\n", doc_def_buf).as_str());
+                        example_description_question_buf.push_str("```\n\n");
+                    }
+                }
+            }
+        }
+    }
+    buf.push_str("## Instructions:\n\n");
+    buf.push_str(
+        format!(
+            "You are an expert in the Roc programming library, `{}`.\n",
+            package_name
+        )
+        .as_str(),
+    );
+    buf.push_str(
+        format!(
+            "Your task is to answer questions **only based on the official documentation** of the `{}` Roc library.\n",
+            package_name
+        )
+        .as_str(),
+    );
+    buf.push_str("If the documentation does not contain relevant information to answer the question, explicitly state: ");
+    buf.push_str(
+        format!(
+            "**I do not know, as this is not covered in the official documentation of `{}`.**\n",
+            package_name
+        )
+        .as_str(),
+    );
+
+    buf.push_str("## Example Responses:\n\n");
+
+    if !example_type_question_buf.is_empty() {
+        buf.push_str(example_type_question_buf.as_str());
+    }
+
+    if !example_description_question_buf.is_empty() {
+        buf.push_str(example_description_question_buf.as_str());
+    }
+
+    buf.push_str("**Gap in Documentation Question Example**\n\n");
+    buf.push_str("**Question:**\n");
+    buf.push_str("What's your favorite animal?\n\n");
+    buf.push_str("**Response:**\n");
+    buf.push_str(
+        format!(
+            "I do not know, as this is not covered in the official documentation of `{}`.\n\n",
+            package_name
+        )
+        .as_str(),
+    );
+
+    buf
+}
```

Comment:
I'd like to leave out these extras for now. I feel like we should just provide the information in an LLM accessible way here, and not try to influence it's system prompt. Similar to how [fasthtml does it](https://docs.fastht.ml/llms-ctx.txt).